scene/2d/screen_button.cpp

```diff
@@ -86,6 +118,15 @@ void TouchScreenButton::_notification(int p_what) {
 					draw_texture(texture,Point2());
 			}
 
+			if (!get_tree()->is_editor_hint() && !get_tree()->is_debugging_collisions_hint())
+				return;
+			if (shape.is_valid()) {
```

Comment:
I've written it that way because that same pair of conditions is checked at other places so that it can be pattern-matched while one is reading the code as: "if there is no reason to show the shape, skip."
That is, keeping that precise check as an unit, instead of mixing it with other surrounding checks for skip along the class.